`sage.graphs.path_enumeration` fails to build in macos-arm64 wheels

related to `PairingHeap`

```
  building 'sage.graphs.path_enumeration' extension
  clang++ -Wsign-compare -Wunreachable-code -fno-common -dynamic -DNDEBUG -g -fwrapv -O3 -Wall -g -arch arm64 -I/Users/runner/work/passagemath/passagemath/src/sage/data_structures -I/private/var/folders/xx/pgt1mtfn3xx3hdys7czt35bh0000gn/T/build-env-awu5hd_p/lib/python3.11/site-packages/cysignals -I/Users/runner/work/passagemath/passagemath/src/sage/cpython -I/Users/runner/work/passagemath/passagemath/src -I/Library/Frameworks/Python.framework/Versions/3.11/include/python3.11 -I/private/var/folders/xx/pgt1mtfn3xx3hdys7czt35bh0000gn/T/build-env-awu5hd_p/include -I/Library/Frameworks/Python.framework/Versions/3.11/include/python3.11 -c sage/graphs/path_enumeration.cpp -o build/temp.macosx-11.0-arm64-cpython-311/sage/graphs/path_enumeration.o
  sage/graphs/traversals.cpp:13883:62: warning: unused function '__pyx_f_4sage_6graphs_4base_21static_sparse_backend_19StaticSparseBackend_cg' [-Wunused-function]
  static struct __pyx_obj_4sage_6graphs_4base_7c_graph_CGraph *__pyx_f_4sage_6graphs_4base_21static_sparse_backend_19StaticSparseBackend_cg(struct __pyx_obj_4sage_6graphs_4base_21static_sparse_backend_StaticSparseBackend *__pyx_v_self) {
                                                               ^
  In file included from sage/graphs/path_enumeration.cpp:1222:
  /Users/runner/work/passagemath/passagemath/src/sage/data_structures/./pairing_heap.h:167:21: error: expected member name or ';' after declaration specifiers
          : value{some_value}, prev{nullptr}, next{nullptr}, child{nullptr} {
                             ^
  /Users/runner/work/passagemath/passagemath/src/sage/data_structures/./pairing_heap.h:167:9: error: expected '('
  clang -bundle -undefined dynamic_lookup -g -L/Users/runner/sage-local/lib -Wl,-rpath,/Users/runner/sage-local/lib -Wl,-ld_classic -Wl,-headerpad_max_install_names -arch arm64 build/temp.macosx-11.0-arm64-cpython-311/sage/graphs/views.o -lgmp -o build/lib.macosx-11.0-arm64-cpython-311/sage/graphs/views.cpython-311-darwin.so
          : value{some_value}, prev{nullptr}, next{nullptr}, child{nullptr} {
                 ^
  /Users/runner/work/passagemath/passagemath/src/sage/data_structures/./pairing_heap.h:167:20: error: expected ';' after expression
          : value{some_value}, prev{nullptr}, next{nullptr}, child{nullptr} {
                            ^
                            ;
  /Users/runner/work/passagemath/passagemath/src/sage/data_structures/./pairing_heap.h:178:60: error: a space is required between consecutive right angle brackets (use '> >')
      : public PairingHeapNodeBase<TV, PairingHeapNode<TI, TV>> {
                                                             ^~
                                                             > >
  /Users/runner/work/passagemath/passagemath/src/sage/data_structures/./pairing_heap.h:192:19: warning: alias declarations are a C++11 extension [-Wc++11-extensions]
  building 'sage.graphs.isoperimetric_inequalities' extension
      using Base_ = PairingHeapNodeBase<TV, PairingHeapNode<TI, TV>>;

  /Users/runner/work/passagemath/passagemath/src/sage/data_structures/./pairing_heap.h:260:37: error: no member named '_merge' in 'pairing_heap::PairingHeapNode<int, double>'
          root = empty() ? p : HeapNodeType::_merge(root, p);
                                             ^
  sage/graphs/path_enumeration.cpp:22978:35: note: in instantiation of member function 'pairing_heap::PairingHeap<int, double>::push' requested here
        __pyx_cur_scope->__pyx_v_pq.push(__pyx_t_4, 0.0);
                                    ^
  In file included from sage/graphs/path_enumeration.cpp:1222:
  /Users/runner/work/passagemath/passagemath/src/sage/data_structures/./pairing_heap.h:182:9: error: type 'Base_' (aka 'PairingHeapNodeBase<double, PairingHeapNode<int, double> >') is not a direct or virtual base of 'pairing_heap::PairingHeapNode<int, double>'
        : Base_(some_value), item(some_item) {
          ^~~~~
  /Users/runner/work/passagemath/passagemath/src/sage/data_structures/./pairing_heap.h:258:40: note: in instantiation of member function 'pairing_heap::PairingHeapNode<int, double>::PairingHeapNode' requested here
        PairingHeapNode<TI, TV> *p = new PairingHeapNode<TI, TV>(some_item, some_value);
                                         ^
  sage/graphs/path_enumeration.cpp:22978:35: note: in instantiation of member function 'pairing_heap::PairingHeap<int, double>::push' requested here
        __pyx_cur_scope->__pyx_v_pq.push(__pyx_t_4, 0.0);
                                    ^
  In file included from sage/graphs/path_enumeration.cpp:1222:
  /Users/runner/work/passagemath/passagemath/src/sage/data_structures/./pairing_heap.h:268:47: error: no member named 'value' in 'pairing_heap::PairingHeapNode<int, double>'
        return std::make_pair(root->item, root->value);
                                          ~~~~  ^
  sage/graphs/path_enumeration.cpp:23017:49: note: in instantiation of member function 'pairing_heap::PairingHeap<int, double>::top' requested here
          __pyx_t_5 = __pyx_cur_scope->__pyx_v_pq.top();
                                                  ^
  In file included from sage/graphs/path_enumeration.cpp:1222:
  /Users/runner/work/passagemath/passagemath/src/sage/data_structures/./pairing_heap.h:326:26: error: no member named 'value' in 'pairing_heap::PairingHeapNode<int, double>'
        return it->second->value;
               ~~~~~~~~~~  ^
  sage/graphs/path_enumeration.cpp:23456:56: note: in instantiation of member function 'pairing_heap::PairingHeap<int, double>::value' requested here
                __pyx_t_11 = __pyx_cur_scope->__pyx_v_pq.value(__pyx_t_4);
                                                         ^
  In file included from sage/graphs/path_enumeration.cpp:1222:
  /Users/runner/work/passagemath/passagemath/src/sage/data_structures/./pairing_heap.h:302:9: error: no member named 'value' in 'pairing_heap::PairingHeapNode<int, double>'
          if (p->value <= new_value) {
              ~  ^
  sage/graphs/path_enumeration.cpp:23479:45: note: in instantiation of member function 'pairing_heap::PairingHeap<int, double>::decrease' requested here
                  __pyx_cur_scope->__pyx_v_pq.decrease(__pyx_t_4, __pyx_t_11);
                                              ^
  In file included from sage/graphs/path_enumeration.cpp:1222:
  /Users/runner/work/passagemath/passagemath/src/sage/data_structures/./pairing_heap.h:305:5: error: no member named 'value' in 'pairing_heap::PairingHeapNode<int, double>'
          p->value = new_value;
          ~  ^
  /Users/runner/work/passagemath/passagemath/src/sage/data_structures/./pairing_heap.h:306:9: error: no member named 'prev' in 'pairing_heap::PairingHeapNode<int, double>'
          if (p->prev != nullptr) {
              ~  ^
  11 warnings and 16 errors generated.
```
https://github.com/passagemath/passagemath/actions/runs/21304553133/job/61330813669#step:21:804

--- a/src/sage/data_structures/pairing_heap.pxd
+++ b/src/sage/data_structures/pairing_heap.pxd
@@ -1,4 +1,7 @@
 # sage_setup: distribution = sagemath-categories
+# distutils: language = c++
+# distutils: extra_compile_args = -std=c++17
+
 # ******************************************************************************
 # Copyright (C) 2024 David Coudert <david.coudert@inria.fr>
 #

--- a/src/sage/data_structures/pairing_heap.pyx
+++ b/src/sage/data_structures/pairing_heap.pyx
@@ -1,6 +1,5 @@
 # sage_setup: distribution = sagemath-categories
 # distutils: language = c++
-# distutils: extra_compile_args = -std=c++11
 r"""
 Pairing Heap
 

--- a/src/sage/graphs/base/c_graph.pyx
+++ b/src/sage/graphs/base/c_graph.pyx
@@ -1,6 +1,5 @@
 # sage_setup: distribution = sagemath-graphs
 # distutils: language = c++
-# distutils: extra_compile_args = -std=c++11
 r"""
 Fast compiled graphs
 

--- a/src/sage/graphs/traversals.pyx
+++ b/src/sage/graphs/traversals.pyx
@@ -1,6 +1,5 @@
 # sage_setup: distribution = sagemath-graphs
 # distutils: language = c++
-# distutils: extra_compile_args = -std=c++11
 r"""
 Graph traversals
 

